File: 1ºAno/Lista3/C/Sala/files/ex10.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int main() {
    for (;;) {
        char *dados = (char*)malloc(10 * sizeof (char));
        int vals[3] = {0, 0, 0};
        for (;;) {
            printf ("Na mesma linha, entre com os valores, respectivamente, referentes a:\n\n1) Divisibilidade por 2 dos termos da sequência: 0 se os termos forem pares e 1 se os valores forem ímpares\n2) Primeiro número da sequência (a1)\n3) Último número da sequência (an)\nPara sair, entre com 0 em todos os valores\n\n");
            fgets(dados, 10, stdin);
            char *pt = strtok(dados, " ");
            int i = 0;
            while (pt != NULL) {
                vals[i] = atoi(pt);
                pt = strtok(NULL, " ");
                ++i;
            }
            if (vals[0] == 0 && vals[1] == 0 && vals[2] == 0) {
                break;
            }
            else if (vals[0] != 0 && vals[0] != 1) {
                printf ("Número(s) inválidos... Tente novamente");
            }
            else {
                break;
            }
        }
        if (vals[0] == 0 && vals[1] == 0 && vals[2] == 0) {
            break;
        }
        if ((vals[0] == 0 && vals[1] % 2 != 0) || (vals[0] == 1 && vals[1] % 2 == 0)) {
            if (vals[1] <= vals[2]) {
                vals[1] += 1;
            }
            else {
                vals[1] -= 1;
            }
        }
        if ((vals[0] == 0 && vals[2] % 2 != 0) || (vals[0] == 1 && vals[2] % 2 == 0)) {
            if (vals[2] > vals[1]) {
                vals[2] -= 1;   
            }
            else {
                vals[2] += 1;
            }
        }
        if (vals[1] <= vals[2]) {
            printf ("A soma dos valores n, tal que n %% 2 = %d, entre %d e %d é %d\n\n", vals[0], vals[1], vals[2], (vals[1] + vals[2]) * ((vals[2] - vals[1]) / 2 + 1) / 2);
        }
        else {
            printf ("A soma dos valores n, tal que n %% 2 = %d, entre %d e %d é %d\n\n", vals[0], vals[1], vals[2], (vals[1] + vals[2]) * ((vals[1] - vals[2]) / 2 + 1) / 2);
        }
    }
    printf ("\nAdeus!");
    return 0;
}

File: 1ºAno/Lista3/Python/Sala/files/ex10.py
while True:
    while True:
        try:
            dados = input("Entre, na mesma linha, com os valores, respectivamente, referentes a:\n\n1) Paridade dos termos da sequência (0 se forem pares e 1 se forem ímpares)\n2) 1º Termo da sequência\n2) Último termo da sequência\nPara sair, entre com 0 em todos os valores\n\n").split()
            dados[0], dados[1], dados[2] = int(dados[0]), int(dados[1]), int(dados[2])
            break
        except ValueError:
            print ("Número(s) inválido(s)... Tente novamente")
    if dados[0] == dados[1] == dados[2] == 0:
        break
    if dados[0] == 0:
        print (f"A soma dos números pares, entre {dados[1]} e {dados[2]} é ", end='')
    else:
        print (f"A soma dos números ímpares, entre {dados[1]} e {dados[2]} é ", end='')
    if dados[0] == 0 and dados[1] % 2 != 0 or dados[0] == 1 and dados[1] % 2 == 0:
        dados[1] += 1
    if dados[0] == 0 and dados[2] % 2 != 0 or dados[0] == 1 and dados[2] % 2 == 0:
        dados[2] -= 1
    print((dados[1] + dados[2]) * ((dados[2] - dados[1]) // 2 + 1) // 2)
print ("\nAdeus")